### PR TITLE
[ASV-858] Refactored notifications handling on installService.

### PR DESCRIPTION
**Pull request by tiagopedrinho**
**What does this PR do?**

   Refactored the notifications handling on InstallService to be according to the rest of the app MVP.

**Database changed?**

  No

**Where should the reviewer start?**

- [x] InstallService.java
- [x] DownloadsNotification.java
- [x] DownloadsNotificationPresenter.java

**How should this be manually tested?**

  Check if the notification flow on the updates tab works as the release.

**What are the relevant tickets?**

  https://aptoide.atlassian.net/browse/ASV-858

**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] If yes - Database migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Partners build
- [ ] Unit tests pass
- [ ] Functional QA tests pass

### CODE REVIEW

File: app/src/main/java/cm/aptoide/pt/install/DownloadsNotificationsPresenter.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            String md5 = installation.getMd5();
+            view.setupNotification(md5, installation.getAppName(), installation.getProgress(), installation.isIndeterminate());
+          }
+        }, throwable -> {
+          Log.e("TAG", "Error on handleOpenDownloadManager");
+          view.removeNotificationAndStop();
+        }));
+  }
+
+  public void onDestroy() {
```
Comment (JDSM01) [inline app/src/main/java/cm/aptoide/pt/install/DownloadsNotificationsPresenter.java]: Not being called

File: app/src/main/java/cm/aptoide/pt/install/InstallService.java [hunk 2]
```diff
@@ -58,8 +59,12 @@
   private CompositeSubscription subscriptions;
   private Notification notification;
   private String marketName;
+  private PublishSubject<String> openAppViewAction;
+  private PublishSubject<Void> openDownloadManagerAction;
```
Comment (JDSM01) [inline app/src/main/java/cm/aptoide/pt/install/InstallService.java]: Subjects are not being destroyed

File: app/src/main/java/cm/aptoide/pt/install/DownloadsNotificationsPresenter.java (hunk 1 above)
Comment (tiagopedrinho) [reply app/src/main/java/cm/aptoide/pt/install/DownloadsNotificationsPresenter.java]: solved.

File: app/src/main/java/cm/aptoide/pt/install/InstallService.java [hunk 3]
```diff
@@ -68,7 +73,10 @@
     installManager = application.getInstallManager();
     marketName = application.getMarketName();
     subscriptions = new CompositeSubscription();
-    setupNotification();
+    openDownloadManagerAction = PublishSubject.create();
+    openAppViewAction = PublishSubject.create();
+    presenter = new DownloadsNotificationsPresenter(this, installManager);
+    presenter.onCreate();
```
[the remark this answers is not in the document]
Comment (tiagopedrinho) [reply app/src/main/java/cm/aptoide/pt/install/InstallService.java]: solved

File: app/src/main/java/cm/aptoide/pt/install/DownloadsNotificationsPresenter.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+import android.util.Log;
+import rx.subscriptions.CompositeSubscription;
+
+/**
+ * Created by tiagopedrinho on 10/10/2018.
+ */
+
+public class DownloadsNotificationsPresenter {
+
+  private DownloadsNotification view;
```
[the remark this answers is not in the document]
Comment (tiagopedrinho) [reply app/src/main/java/cm/aptoide/pt/install/DownloadsNotificationsPresenter.java]: solved

File: app/src/main/java/cm/aptoide/pt/install/InstallService.java (hunk 2 above)
Comment (tiagopedrinho) [reply app/src/main/java/cm/aptoide/pt/install/InstallService.java]: solved.

File: app/src/main/java/cm/aptoide/pt/install/DownloadsNotificationsPresenter.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+      InstallManager installManager) {
+    this.view = view;
+    this.installManager = installManager;
+    subscriptions = new CompositeSubscription();
+  }
+
+  public void onCreate() {
+    handleOpenAppView();
+    handleOpenDownloadManager();
+    getCurrentInstallation();
```
[the remark this answers is not in the document]
Comment (tiagopedrinho) [reply app/src/main/java/cm/aptoide/pt/install/DownloadsNotificationsPresenter.java]: solved

**Pull request merged**